src/window/linux/CMakeLists.txt

```
@@ -0,0 +1,30 @@
+set(NOX_WINDOW_LINUX_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt
+    ${CMAKE_CURRENT_SOURCE_DIR}/linux_window.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/linux_window.h
+)
+
+target_sources(${NOX_LIB_NAME} PRIVATE ${NOX_WINDOW_LINUX_SOURCES})
+
+set(NOX_IMPL_WINDOW_LINUX_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/CMakeLists.txt
+    ${CMAKE_CURRENT_SOURCE_DIR}/linux_window_helper.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/linux_window_helper.h
+)
+
+
+target_sources(${NOX_IMPL_LIB_NAME} PRIVATE ${NOX_IMPL_WINDOW_LINUX_SOURCES})
+
+# --------------------------------------------------
+# Link libraries
+# --------------------------------------------------
+
+find_package(X11 4.6 REQUIRED)
+
+if(NOT X11_FOUND)
+    message(FATAL_ERROR "X11 not found")
+endif()
+
+include_directories(${X11_INCLUDE_DIR})
+link_directories(${X11_LIBRARIES})
+target_link_libraries(${NOX_IMPL_LIB_NAME} PRIVATE ${X11_LIBRARIES})
```

```suggestion
target_include_directories(
    ${NOX_IMPL_LIB_NAME} 
    PRIVATE
    "$<BUILD_INTERFACE:${X11_INCLUDE_DIR}>"
)
target_link_libraries(
    ${NOX_IMPL_LIB_NAME}
    PRIVATE
    ${X11_LIBRARIES}
)
```

`include_directories` and `link_directories` are legacy functions, `target_include_directories` and `target_link_libraries` are preferred.